### PR TITLE
[prometheus-blackbox-exporter] Add custom relabeling and quote values

### PR DESCRIPTION
**Pull request by epitstickpan**
Adds `additionalRelabeling` to ServiceMonitor so that custom relabeling
can be done. I wanted this because I wanted to be able to add the module
as a label but I think there are other good reasons to have it.

While I was here I added `quote` for `additionalMetricsRelabels` so that
values wouldn't be converted.

Please make note of the fact I did a concat instead of replacing default values. I think this is more productive but it works differently than `additionalMetricsRelabels`. Let me know if you want me to change one or the other so they work the same. Wanted to avoid a major bump if I could.

@desaintmartin 

Signed-off-by: Evan Pitstick <epitstick@paloaltonetworks.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-blackbox-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 description: Prometheus Blackbox Exporter
 name: prometheus-blackbox-exporter
-version: 5.7.0
+version: 5.7.1
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/Chart.yaml:4]: new feature should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-blackbox-exporter/Chart.yaml (hunk 1 above)
Comment (epitstickpan) [reply charts/prometheus-blackbox-exporter/Chart.yaml]: Thanks, got it.

**Review by monotek — APPROVED** [review]

**Comment by Gentoli** [comment]
This causes a panic from helm when `target.additionalRelabeling` is not present:
```
Error: template: prometheus-blackbox-exporter/templates/servicemonitor.yaml:43:7: executing "prometheus-blackbox-exporter/templates/servicemonitor.yaml" at <concat .additionalRelabeling $.Values.serviceMonitor.defaults.additionalRelabeling>: error calling concat: runtime error: invalid memory address or nil pointer dereference
```

This is probably a helm issues, but right now it's broken when `target.additionalRelabeling` to not set.

helm version:
`version.BuildInfo{Version:"v3.8.2", GitCommit:"6e3701edea09e5d55a8ca2aae03a68917630e91b", GitTreeState:"clean", GoVersion:"go1.18.1"}`

**Comment by epitstickpan** [comment]
> This causes a panic from helm when `target.additionalRelabeling` is not present:
> 
> ```
> Error: template: prometheus-blackbox-exporter/templates/servicemonitor.yaml:43:7: executing "prometheus-blackbox-exporter/templates/servicemonitor.yaml" at <concat .additionalRelabeling $.Values.serviceMonitor.defaults.additionalRelabeling>: error calling concat: runtime error: invalid memory address or nil pointer dereference
> ```
> 
> This is probably a helm issues, but right now it's broken when `target.additionalRelabeling` to not set.
> 
> helm version: `version.BuildInfo{Version:"v3.8.2", GitCommit:"6e3701edea09e5d55a8ca2aae03a68917630e91b", GitTreeState:"clean", GoVersion:"go1.18.1"}`

I have a fix coming in for it. Thanks for the heads up.

**Pull request merged**